src/shared/cloudformation/cloudformation.ts

```diff
@@ -494,18 +502,47 @@ export namespace CloudFormation {
         return matches[0]
     }
 
-    function matchesHandler({ resource, handlerName }: { resource?: Resource; handlerName: string }) {
-        return (
-            resource &&
-            resource.Type === SERVERLESS_FUNCTION_TYPE &&
-            isZipLambdaResource(resource.Properties) &&
-            // TODO: `resource.Properties.Handler` is relative to `CodeUri`, but
-            //       `handlerName` is relative to the directory containing the source
-            //       file. To fix, update lambda handler candidate searches for
-            //       interpreted languages to return a handler name relative to the
-            //       `CodeUri`, rather than to the directory containing the source file.
-            resource.Properties.Handler === handlerName
-        )
+    /**
+     * Gets the resource with a `Events.*.Path` field matching the given API path.
+     *
+     * Duplicate API paths are resolved as "last wins", which is compatible with SAM.
+     *
+     * Example (path "/hello" matches "HelloWorldFunction"):
+     *
+     *     HelloWorldFunction:
+     *       Type: AWS::Serverless::Function
+     *       Properties:
+     *         PackageType: Image
+     *         Events:
+     *           HelloWorld:
+     *             Type: Api
+     *             Properties:
+     *               Path: /hello
+     *               Method: get
+     */
+    export function getResourceByApiPath(resources: TemplateResources, apiPath: string): string {
```

Comment:
Nice catch! I also forgot that `target=template` [now uses](https://github.com/aws/aws-toolkit-vscode/pull/2559#discussion_r842128545) the real template.yaml instead of a partial temporary one! So we probably don't need `getResourceByApiPath()`.